lib/auth.js

```
@@ -103,5 +104,28 @@ exports.loginRequired = function(req, res, next) {
     if (req.loggedIn) {
         return next();
     }
-    res.redirect('/login');
-};
\ No newline at end of file
+    res.redirect(everyauth.password.getLoginPath() +
+            "?next=" + encodeURIComponent(req.url));
+};
+
+exports.respondToLoginSucceed = function(res, user, data) {
+    if(!user) return;
+
+    var loc = '/';
+    if(data.req.body.next) {
+        loc = data.req.body.next;
+    }
+    res.writeHead(303, {'Location': loc});
+    res.end();
+};
+
+exports.respondToRegistrationSucceed = function(res, user, data) {
+    if(!user) return;
```

Same with registration, but we should maintain the inputs that were given (except password). This may be grounds for using [express-form](https://github.com/dandean/express-form), if it works anything like Django's forms.